### PR TITLE
DOC-646 Document AWS BYOC prereqs

### PR DESCRIPTION
**Pull request by micheleRP**
## Description
This adds a prereqs section to the doc for BYOC on AWS (based on [this slack thread](https://redpandadata.slack.com/archives/C0191NM4WU9/p1729002110977339)).

Resolves https://redpandadata.atlassian.net/browse/DOC-646
Review deadline: Friday Oct 25

## Page previews
[Create a BYOC Cluster on AWS](https://deploy-preview-96--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-aws/)

## Checks

- [ ] New feature
- [X] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by voutilad — APPROVED** [review]
Looks good to me.

File: modules/get-started/pages/cluster-types/byoc/aws/create-byoc-cluster-aws.adoc [hunk 1]
```diff
@@ -6,6 +6,21 @@ To create a Redpanda cluster in your virtual private cloud (VPC), follow the ins
 
 See also: xref:get-started:cloud-overview.adoc#redpanda-cloud-architecture[Redpanda Cloud architecture].
 
+== Prerequisites
+
+With BYOC, Redpanda manages security policies and resources for your VPC, including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. Before you deploy a BYOC cluster on AWS, check that the user creating the cluster has the following prerequisites: 
+
+* A minimum version of Redpanda `rpk` v24.1. See xref:manage:rpk/rpk-install.adoc[].
+* The permissions necessary to launch infrastructure for running Redpanda. See xref:security:authorization/cloud-iam-policies.adoc[IAM policies]
```
Comment (kbatuigas) [inline modules/get-started/pages/cluster-types/byoc/aws/create-byoc-cluster-aws.adoc:14]:
```suggestion
* The permissions necessary to launch infrastructure for running Redpanda. See xref:security:authorization/cloud-iam-policies.adoc[IAM policies].
```
nit, for consistency

File: modules/security/partials/iam-policies.adoc [hunk 2]
```diff
@@ -4,7 +4,7 @@ agent access, so that brokers can communicate with the BYOC clusters.
 
 See also: xref:get-started:cloud-overview.adoc#byoc-architecture[BYOC architecture]
 
-NOTE: This page lists the IAM permissions Redpanda uses. Nothing is required by Redpanda Cloud users. 
+NOTE: This page lists the IAM permissions Redpanda needs to create BYOC clusters. Nothing is required by Redpanda Cloud users.
```
Comment (kbatuigas) [inline modules/security/partials/iam-policies.adoc:7]: To clarify: did we mean "required of/for"?

**Review by kbatuigas — APPROVED** [review]
No blockers, just a tiny edit and a clarifying question.

**Pull request merged**